aomi/model/auth.py

```diff
@@ -232,6 +226,73 @@ def delete(self, client):
         client.delete_role(self.app_name)
 
 
+class LDAP(Auth):
+    """LDAP Authentication"""
+    required_fields = ['url']
+    config_key = 'ldap_auth'
+
+    def __init__(self, obj, opt):
+        super(LDAP, self).__init__('ldap', obj, opt)
+        auth_obj = {
+            'url': obj['url']
+        }
+        self.mount = 'ldap'
+        self.path = sanitize_mount("auth/ldap/config")
+        map_val(auth_obj, obj, 'starttls', False)
```

Comment:
why do these mutate in place rather then passing out new object?
then you might do something fun like
```python 
reduce(lambda key, auth_obj: map_val(auth_obj,obj,key))
```
and have the keys be data driven